### PR TITLE
Fix reading and ansi rendering of bold, italic, underline, strikethrough

### PR DESCRIPTION
**Pull request by knivey**

### CODE REVIEW

File: ascii/color.go [hunk 1]
```diff
@@ -51,31 +52,40 @@ func (c ColorPair) MircCode(p *Palette) []byte {
...
 		return ret
 	}
-	ret = append(ret, []byte("\u001b[4")...)
-	ret = append(ret, ansiCode(c.Background)...)
+	if code, high := ansiCode(c.Background); !high {
+		ret = append(ret, append([]byte("\u001b[4"), code...)...)
+	} else {
+		ret = append(ret, append([]byte("\u001b[10"), code...)...)
+	}
 	if string(ret[len(ret)-2:]) == ";1" {
```
Comment (chzchzchz) [inline ascii/color.go:89]: this code block can be removed because `ansiCode` changed

File: ascii/ascii.go [hunk 2]
```diff
@@ -59,7 +71,7 @@ func NewASCII(dat string) (*ASCII, error) {
 			if fg == -1 {
 				// Reset color codes.
 				fg, bg = -1, -1
-				continue
+				break
```
Comment (chzchzchz) [inline ascii/ascii.go:74]: add test case in ascii_test.go? I don't understand what this particular change fixes

**Review by chzchzchz — CHANGES_REQUESTED** [review]
ty knivey

File: ascii/ascii.go (hunk 2 above)
Comment (knivey) [reply ascii/ascii.go:74]: without the break it would eat a character after a \x03 to reset colors because it wasn't a number

**Pull request merged**